### DISCUSSION TITLE
Adding build steps to an automatic now deployment

### DISCUSSION THREAD

**Discussion by ajhool** [vercel/vercel#4198] [General] [answered]
Current tooling:

- next.js
- now.sh for automatic deployments on push. Just learning today that now has been renamed vercel.
- AWS Amplify CLI + js
- typescript
- gitlab

At some point in my CI platform (and locally) we need to make a call to AWS Amplify to do code generation based on the build environment. For example, in a production CI, `amplify cli` needs to be run: `amplify pull --env prod --appId some-id` to do codegen for the production deployment. Similar commands need to be run for `env staging` and `env some-feature-branch`. The env is set in environment variables and does not require code changes(!).

We are struggling to figure out how to inject this build step into either `now` or `next`. The Amplify CLI is a node module but is called as a command line executable.

**Comment by paulogdm** [comment, marked as answer]
Closing in favor of #4199
